src/wp-admin/update-core.php

```diff
@@ -680,6 +680,13 @@ function list_theme_updates() {
 
 		$compat = '';
 
+		// Get the upgrade notice for the new theme version.
+		if ( isset( $theme->update['upgrade_notice'] ) ) {
+				$upgrade_notice = '<br/>' . strip_tags( $theme->update['upgrade_notice'] );
```

Comment:
If this isn't used until line 781, what about just adding this to line 781?

```php
// Get the upgrade notice for the new theme version.
if ( isset( $theme->update['upgrade_notice'] ) ) {
	echo '<br/>' . strip_tags( $theme->update['upgrade_notice'] );
}
```

Reply:
Also, is it possible that `$theme->update['upgrade_notice']` could ever be set to `false`? If so, would this be safer?

```php
if ( isset( $theme->update['upgrade_notice'] ) && is_string( $theme->update['upgrade_notice'] ) ) {
```

Reply:
As it's parses from the readme.txt I don't believe it could ever be `false`. If present, it will be text.

Reply:
Cheers for confirming that Andy. What are your thoughts on [this comment](https://github.com/WordPress/wordpress-develop/pull/4238#discussion_r1233448847) above?

Reply:
Probably correct. I'll check all the stuff tomorrow. As for why it structured like it is, take a look at the PR for plugins referenced in the ticket. I think the structure started there. 

And yes, I'm just trying to do that from memory. 😂

Reply:
After checking this is the structure used for plugins and making these 2 sections somewhat equivalent was the goal.